source/dub/project.d

```diff
@@ -473,6 +473,11 @@ class Project {
 				Dependency vspec = dep.spec;
 				Package p;
 
+				if (vspec.optional && vspec.path() != NativePath.init && !existsDirectory(vspec.path())) {
+					logDiagnostic("%sOptional dependency %s path %s does not exist, not using.", indent, dep.name, vspec.path());
```

Comment:
I think this should rather fall under a warning, since when typo-ing the path, the user won't understand why the modules they import aren't being found and probably first try to search in the dependency.

Reply:
But on the other hand, do we really want Dub to print a warning every time you build a project even when nothing is wrong and everything is as intended and there is nothing to fix?

Reply:
well imo that's better than no warning for such issues

Reply:
Rule of silence???

Reply:
I would agree if this would be a warning that the user can't get rid of, however you can remove it by fixing the path (what people typoing usually want) or currently just commenting out such dependencies when they are optional. However I think we can have a better solution than this, like a new setting per dependency.

I'm not a fan of reusing optional, since users might just manually set a path in existing dependencies that are optional. While it's unlikely, I don't really see a major reason not to add a new key for this.

Reply:
Alright.